scripts/source-maintenance.sh

```diff
@@ -35,6 +35,14 @@ while [ $# -ge 1 ]; do
         KeepGoingDirective=$1
         shift
         ;;
+    --skip-astyle-version-check)
```

Comment:
We already protect sources via a `git diff` check. A very wrong astyle version cannot really do any serious damage, right?

IMO, we should convert the existing version check into a strong warning instead of preserving it while adding a bypass option. IIRC, your own investigations showed that several astyle versions produce identical results for official sources. With git protections, there is no need for this annoying and unusual check anymore.

You will probably need Amos blessing on this. If you get it, this change requests overwrites other conflicting change requests in my review.

Reply:
Not directly. But it could produce a huge diff that someone not being careful enough commits - eg on a branch with already large logic changes.

Reply:
resolving conversation, this is solved via --enable-astyle

Reply:
> Alex: A very wrong astyle version cannot really do any serious damage, right?

> Amos: it could produce a huge diff that someone not being careful enough commits - eg on a branch with already large logic changes.

And that huge commit on a huge branch is _not_ a major damage. It is trivial to revert (or effectively undo by reformatting). Yes, there will always be esoteric ways to shoot oneself in the foot, but we have more than adequate protection here already.

It is sad that we continue to work around problems introduced by an obsolete (at best) feature instead of simply removing that feature, but I do not have the time to fight this.